src/era/rotating.py

```diff
@@ -0,0 +1,106 @@
+        self._class_logger.debug(f'Folder regex pattern: {pattern}')
+        folder_paths = [path for path in self.base_dir.iterdir() if re.match(pattern, path.stem)]
+        if max_folder_count > len(folder_paths):
+            self._class_logger.debug(
+                f'max_folder_count `{max_folder_count}` greater than number of existing folders `{len(folder_paths)}`; '
+                'no folders deleted'
+            )
+            max_folder_count = 0
+        return sorted(folder_paths)[:-max_folder_count]
+
+    def _delete_old_folders(self, folder_paths_to_delete):
+        deleted_folders = []
+        for folder in folder_paths_to_delete:
+            try:
+                self._class_logger.debug(f'Attempting to delete `{folder}`')
+                shutil.rmtree(folder)
+            except Exception:
+                pass
+            else:
+                self._class_logger.debug(f'Successfully deleted `{folder}`')
+                deleted_folders.append(folder)
+
+        return deleted_folders
+
+    def get_rotated_directory(
+        self,
+        prefix='',
+        date_format='%Y%m%d_%H%M%S',
+        exist_ok=False,
+        pattern='[0-9]{8}_[0-9]{6}',
+        max_folder_count=10
+    ):  # pylint: disable=too-many-arguments
+        """Get a new directory using the info provided, rotating according to max_folder_count.
+
+        Logs to logging.getLogger(__name__).getChild(self.__class__.__name__).
+        DEBUG:
+            Directory creation attempt/success
+            Individual directory deletion attempts/successes
+        INFO:
+            Directories deleted as part of rotation
+
+        Raises FileNotFoundError if base_dir does not exist. Silently swallows any errors when deleting folders
+        and moves on to the next folder to be deleted.
+
+        Args:
+            prefix (str, optional): Folder prefix for date. Any spacers(_, -, etc) must be provided in the prefix.
+                Defaults to ''.
+            date_format (str, optional): Date format argument to strftime(). Defaults to '%Y%m%d_%H%M%S'
+                (YYYYMMDD_hhmmss).
+            exist_ok (bool, optional): Overwrite an existing folder?. Defaults to False.
+            pattern (str, optional): Regex date matching pattern to determine folders eligible for rotation deletion.
+                Defaults to '[0-9]{8}_[0-9]{6}' (matches default date_format).
+            max_folder_count (int, optional): Number of folders to keep, not including the folder created.
+                Defaults to 10.
+
+        Returns:
+            Path: Path object to newly created folder
+        """
+        folder_paths_to_delete = self._get_all_but_n_most_recent_folder_paths(prefix, pattern, max_folder_count)
+        deleted_folders = self._delete_old_folders(folder_paths_to_delete)
+        self._class_logger.info(
+            f'Deleted {len(deleted_folders)} folder(s) for rotation: {[str(folder) for folder in deleted_folders]}'
```

Comment:
**suggestion**: using the `folder.name` might make the logs a little less dense since the `base_dir` is static

Reply:
It would, but I'm inclined to stay with the full path because it explicitly tells you which folders were deleted, with no room for misinterpretation or hidden bugs where somehow `base_dir` changes to something the user wasn't expecting.